### DISCUSSION TITLE
Automatically running mosaic scenario multiple times

### DISCUSSION THREAD

**Discussion by pv2-0** [eclipse/mosaic#69] [Q&A]
Hi,

I want to run multiple trials of each scenario with different sets of random trips. The only way that I know of for generating random trips is the --generate-trips flag when creating the scenario. However, I've edited both the mapping_config.json file and the application folder. 

How can I generate a new set of random trips without having to add in my applications manually every time?Also, is there a way to automatically assign some trips to one vehicle prototype and the rest of the trips to another vehicle prototype?  It wouldn't be feasible to do these things manually for each trial.

Thanks

**Comment by kschrab** [comment]
Hi, we currently do not provide any tools tackling your problem. Most of the time we build small scripts (e.g. Java or python) to generate or manipulate mapping_config files specifically for what we want to achieve. For a more sophisticated way to calculate routes and configure traffic demand, we have developed a tool called TrafficGen, but due to internal policies we can only provide this tool on a commercial basis. 

However, you could have a look into the tooling of SUMO. They have a loot of python scripts to generate different sets of routes. Furthermore, you could use the concept of `typeDistributions` in the mapping to minimize editing effort. Here you could configure globally that a proportion of all trips are assigned to vehicle type x and all other trips are assigned to vehicle type y. You can [read here in our documentation](https://www.eclipse.org/mosaic/docs/simulators/application_mapping/#use-type-distributions-in-complex-traffic-scenarios) how to do that.
